packages/components/multiselect/src/Multiselect.test.tsx

```diff
@@ -51,20 +59,26 @@ function renderComponent(customProps: Partial<MultiselectProps>) {
       })}
     </Multiselect>,
   );
-}
-
+  return [{ user }] as [{ user: typeof user }];
+};
 describe('Multiselect basic usage', () => {
-  it('opens and closes the drawer', () => {
-    renderComponent({});
-    userEvent.click(screen.getByRole('button', { name: 'Toggle Multiselect' }));
-    expect(screen.getByRole('list')).toBeInTheDocument();
-    userEvent.click(screen.getByRole('button', { name: 'Toggle Multiselect' }));
+  it('opens and closes the drawer', async () => {
+    const [{ user }] = renderComponent();
+
+    const toggleButton = screen.getByRole('button', {
+      name: 'Toggle Multiselect',
+    });
+    await waitFor(() => user.click(toggleButton));
```

Comment:
```suggestion
    await user.click(toggleButton);
```

Reply:
the test setup here seems different, if i do not use waitFor, the tests will not finish and get stuck.